### ISSUE TITLE
Missing prebuilt binaries for musl

### ISSUE THREAD

**Issue by orgads** [atom/node-keytar#302]
* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

The prebuilt binaries don't include musl build. This is nowadays one of the most common distribution of node, since it is used in the `alpine` docker, which is the smallest image of node.

### Steps to Reproduce

```
docker run -it node:12.18.3-alpine sh
mkdir /test
cd /test
npm init -y
npm i keytar
> keytar@6.0.1 install /test/node_modules/keytar
> prebuild-install || node-gyp rebuild

prebuild-install WARN install No prebuilt binaries found (target=12.18.3 runtime=node arch=x64 libc=musl platform=linux)
```

Please include prebuilt binaries also for linuxmusl. Thanks.

**Comment by shiftkey** [comment]
I'd love to add support for this but it requires reworking a bunch of infrastructure to support it - and I don't have the time or energy for that currently.

In the meantime, you should be able to support falling back to `node-gyp` and building locally by adding these packages:

```dockerfile
FROM node:alpine

RUN apk add --no-cache --virtual python make g++ pkgconf libsecret-dev
RUN npm install keytar
```

**Comment by orgads** [comment]
Here is an example using github actions: https://github.com/sass/node-sass/pull/2823/files

I'll try to create a similar pull request here.

**Comment by shiftkey** [comment]
> I'll try to create a similar pull request here.

At the moment we're on Appveyor and Travis which handles publishing the release to GitHub. While I appreciate the PR as a starting point, it's going to take some time to standardize on Actions.

**Comment by orgads** [comment]
Isn't it possible to use both?

**Comment by shiftkey** [comment]
> Isn't it possible to use both?

I'd prefer to use this chance to standardize on one, rather than grow to three different CI providers generating prebuild artifacts.

**Comment by sschoebinger** [comment]
If I fall back to node-gyp as you described, I still get warnings during compilation which I don't get with debian images.

```#8 86.98 > keytar@5.6.0 install /app/node_modules/keytar
#8 86.98 > prebuild-install || node-gyp rebuild
#8 86.98 
#8 87.91 prebuild-install WARN install No prebuilt binaries found (target=14.15.1 runtime=node arch=x64 libc=musl platform=linux)
#8 91.98 make: Entering directory '/app/node_modules/keytar/build'
#8 91.98   CXX(target) Release/obj.target/keytar/src/async.o
#8 93.34 In file included from ../src/async.cc:4:
#8 93.34 ../../nan/nan.h: In function 'void Nan::AsyncQueueWorker(Nan::AsyncWorker*)':
#8 93.34 ../../nan/nan.h:2294:62: warning: cast between incompatible function types from 'void (*)(uv_work_t*)' {aka 'void (*)(uv_work_s*)'} to 'uv_after_work_cb' {aka 'void (*)(uv_work_s*, int)'} [-Wcast-function-type]
#8 93.34  2294 |     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
#8 93.34       |                                                              ^
#8 94.30   CXX(target) Release/obj.target/keytar/src/main.o
#8 95.35 In file included from ../src/main.cc:1:
#8 95.35 ../../nan/nan.h: In function 'void Nan::AsyncQueueWorker(Nan::AsyncWorker*)':
#8 95.35 ../../nan/nan.h:2294:62: warning: cast between incompatible function types from 'void (*)(uv_work_t*)' {aka 'void (*)(uv_work_s*)'} to 'uv_after_work_cb' {aka 'void (*)(uv_work_s*, int)'} [-Wcast-function-type]
#8 95.35  2294 |     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
#8 95.35       |                                                              ^
#8 96.34   CXX(target) Release/obj.target/keytar/src/keytar_posix.o
#8 97.05 ../src/keytar_posix.cc: In function 'keytar::KEYTAR_OP_RESULT keytar::SetPassword(const string&, const string&, const string&, std::string*)':
#8 97.05 ../src/keytar_posix.cc:38:9: warning: missing sentinel in function call [-Wformat=]
#8 97.05    38 |     NULL);                              // End of arguments.
#8 97.05       |         ^
#8 97.05 ../src/keytar_posix.cc: In function 'keytar::KEYTAR_OP_RESULT keytar::GetPassword(const string&, const string&, std::string*, std::string*)':
#8 97.05 ../src/keytar_posix.cc:61:9: warning: missing sentinel in function call [-Wformat=]
#8 97.05    61 |     NULL);                              // End of arguments.
#8 97.05       |         ^
#8 97.06 ../src/keytar_posix.cc: In function 'keytar::KEYTAR_OP_RESULT keytar::DeletePassword(const string&, const string&, std::string*)':
#8 97.06 ../src/keytar_posix.cc:88:9: warning: missing sentinel in function call [-Wformat=]
#8 97.06    88 |     NULL);                              // End of arguments.
#8 97.06       |         ^
#8 97.06 ../src/keytar_posix.cc: In function 'keytar::KEYTAR_OP_RESULT keytar::FindPassword(const string&, std::string*, std::string*)':
#8 97.06 ../src/keytar_posix.cc:112:9: warning: missing sentinel in function call [-Wformat=]
#8 97.06   112 |     NULL);                              // End of arguments.
#8 97.06       |         ^
#8 97.62   SOLINK_MODULE(target) Release/obj.target/keytar.node
#8 97.71   COPY Release/keytar.node
#8 97.72 make: Leaving directory '/app/node_modules/keytar/build'
```

**Comment by shiftkey** [comment]
I've published `v7.6.0` to NPM which contains prebuild binaries for Alpine Linux, so hopefully this issue is resolved!
